### PR TITLE
Run sanitizers in CI

### PR DESCRIPTION
**Pull request by spacey-sooty**

### CODE REVIEW

**Comment by calcmogul** [comment]
This doesn't actually run the sanitizers yet. The sanitizers are runtime instrumentation, so you have to actually run the tests and examples to catch any bugs.

File: .github/workflows/sanitizers.yml [hunk 1]
```diff
@@ -50,3 +50,6 @@ jobs:
           sudo update-alternatives --install /usr/bin/g++ g++ /usr/bin/g++-14 200
 
       - run: cmake -B build -S . -DBUILD_EXAMPLES=ON ${{ matrix.cmake-args }} -DOPTIMIZER_BACKEND=${{ matrix.optimizer_backend }}
+      - run: cmake --build build --config RelWithDebInfo --parallel $(nproc)
+      - run: ctest --test-dir build -C RelWithDebInfo --output-on-failure
+      - run: cmake --install build --config RelWithDebInfo
```
Comment (calcmogul) [inline .github/workflows/sanitizers.yml:55]: The install step isn't needed for sanitizer runs.

File: .github/workflows/sanitizers.yml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+            cmake-args: "-DCMAKE_BUILD_TYPE=Ubsan"
+            optimizer_backend: sleipnir
+            os: ubuntu-24.04
+          - sanitizer-name: asan with casadi
+            cmake-args: "-DCMAKE_BUILD_TYPE=Asan"
+            optimizer_backend: casadi
+            os: ubuntu-24.04
+          - sanitizer-name: tsan
+            cmake-args: "-DCMAKE_BUILD_TYPE=Tsan"
+            optimizer_backend: sleipnir with casadi
```
Comment (calcmogul) [inline .github/workflows/sanitizers.yml:34]: These backend names seem sus.

Comment (spacey-sooty) [reply .github/workflows/sanitizers.yml:34]: whoops

File: .github/workflows/sanitizers.yml [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+          - sanitizer-name: tsan with casadi
+            cmake-args: "-DCMAKE_BUILD_TYPE=Tsan"
+            optimizer_backend: casadi
+            os: ubuntu-24.04
+          - sanitizer-name: ubsan with casadi
+            cmake-args: "-DCMAKE_BUILD_TYPE=Ubsan"
+            optimizer_backend: casadi
+            os: ubuntu-24.04
+
+    name: "${{ matrix.sanitizer-name }}"
```
Comment (calcmogul) [inline .github/workflows/sanitizers.yml:41]: You can make sanitizer-name just be the name of the sanitizer and append the two variables like this.
```suggestion
    name: "${{ matrix.sanitizer-name }} with ${{ matrix.optimizer_backend }}"
```

File: .github/workflows/sanitizers.yml [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+            optimizer_backend: casadi
+            os: ubuntu-24.04
+          - sanitizer-name: tsan
+            cmake-args: "-DCMAKE_BUILD_TYPE=Tsan"
+            optimizer_backend: casadi
+            os: ubuntu-24.04
+          - sanitizer-name: ubsan
+            cmake-args: "-DCMAKE_BUILD_TYPE=Ubsan"
+            optimizer_backend: casadi
+            os: ubuntu-24.04
```
Comment (calcmogul) [inline .github/workflows/sanitizers.yml:39]: You can just harcode the os in the runs-on line because it'll always be the same for all of them.

**Review by calcmogul — APPROVED** [review]

**Pull request merged**